syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
+                    gateway=self._gateways[ip])
+        except Exception as ex:
+            raise RuntimeError(
+                f'Could not set configuration remote shutter{"group" if is_shutter_group else ""} {ip}/{remote_id}: {ex}')
+
+    def update_remote_shutter_state(self, ip, remote_id, state, reverse, is_shutter_group):
+        try:
+            do_call = True
+            if state == "stopped":
+                remote_state = "stop"
+            elif state == "going_up":
+                remote_state = "up" if not reverse else "down"
+            elif state == "going_down":
+                remote_state = "up" if reverse else "down"
+            else:
+                remote_state = ""
+                do_call = False
+
+            if is_shutter_group:
+                group = "group_"
+            else:
+                group = ""
+
+            if do_call:
+                self._call_remote(api_call=f"do_shutter_{group}{remote_state}", params={
+                    "id": remote_id
+                }, gateway=self._gateways[ip])
+        except Exception as ex:
+            raise RuntimeError(
+                f'Could not set state of remote shutter{"group" if is_shutter_group else ""} {ip}/{remote_id}: {ex}')
 
     @background_task
     def run(self):
-        previous_values = {}
         while True:
             if not self._enabled:
                 time.sleep(30)
                 continue
             try:
                 # Sync sensor values:
-                data_humidities = json.loads(self.webinterface.get_sensor_humidity_status())
-                data_temperatures = json.loads(self.webinterface.get_sensor_temperature_status())
-                if data_humidities['success'] is True and data_temperatures['success'] is True:
-                    for sensor_id in range(len(data_temperatures['status'])):
-                        if sensor_id not in self._sensor_mapping:
-                            continue
-                        data = {'sensor_id': self._sensor_mapping[sensor_id]}
-                        humidity = data_humidities['status'][sensor_id]
-                        if humidity != 255:
-                            data['humidity'] = humidity
-                        temperature = data_temperatures['status'][sensor_id]
-                        if temperature != 95.5:
-                            data['temperature'] = temperature
-                        previous = previous_values.setdefault(sensor_id, {})
-                        if previous.get('temperature') != data.get('temperature') or \
-                                previous.get('humidity') != data.get('humidity'):
-                            previous_values[sensor_id] = data
-                            self._call_remote('set_virtual_sensor', params=data)
+                for gateway in self._gateways.values():
+                    sensor_mapping = gateway.get('sensor_mapping')
+                    if sensor_mapping != {}:
```

Comment:
create less indents
could do if sensor_mapping empty: continue

Reply:
Nice way to get less indents indeed, did not know that yet, thanks!